Driver for MCC USB-3114 DAQ

pieterjanstas opened the pull request:
Driver for MCC USB-3114 DAQ
![IMG_6897](https://github.com/lukingroup/pylabnet/assets/79099250/a8d87527-888e-42a8-9228-cddea908580c)

Contains 16 analog outputs (0-10 V) and 8 digital outputs (Low: 0 V, High: 5 V).
Currently the only functions implemented are set analog voltage and set digital pin high or low. More functions can be added later on from https://github.com/mccdaq/mcculw

The device is also integrated with staticline and automatically uses analog or digital ports depending on specification in config file. Analog and digital pin names are read from dio_assignment_global.json
staticline_example.json
![image](https://github.com/lukingroup/pylabnet/assets/79099250/9659185a-937e-4871-b59d-d23f81e6a49a)
![image](https://github.com/lukingroup/pylabnet/assets/79099250/440052f1-4016-41f3-9b1b-cbca24393cb5)

dio_assignment_global.json (first 3 entries are for other device, ignore)
![image](https://github.com/lukingroup/pylabnet/assets/79099250/ce77c95b-e075-4d6c-9588-fc63260c2fbd)

huanyanqi commented:
- Add comments about instacal and `ignore_instacal` (I think you mentioned it needs to be installed, also there's a command `ignore_instacal`, not sure what that does?)
- Line 64 from `hardware/mcc_usb_daq/mcc_usb_3114.py`, should be 0V instead of 10V
- `pylabnet_logs` should not be pushed to the repo. It seems that the `pylabnet_logs` are located within the repo for a particular computer; in that case you should add this to `.git/info/exclude` instead.

pieterjanstas commented:
- corrected comment about 0 V to 10 V
- removed pylabnet_logs folder

for the install requirement, there is a comment on the top of hardware/mcc_usb_daq/mcc_usb_3114.py: Make sure InstaCal is installed to access the device. This is the only extra requirement to access the device.

I used ignore_instacal() as it was recommended in mcculw examples. It seems it prevents conflicts with previous setting of DAQ parameters. From mcculw documentation:

ignore_instacal():
    """Prevents the Universal Library from automatically adding a DAQ device that has been stored
    in the cb.cfg file by InstaCal. This function must be the first Universal Library function
    invoked in the application. Devices can then be added and configured at runtime using the
    device discovery features. Refer to the "InstaCal, API Detection, or Both?" section of the
    Universal Library User's Guide for additional information.
    """

huanyanqi commented:
Right exactly, can you include a summary of what you wrote as a comment above the `ignore_instacal` line? Otherwise to someone reading it might seem like instacal is "ignored/not needed".
